code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+      });
+      const baseAssetFundsCurrentlyInPoolsBeforeTx = await api.query.tokens.accounts(
+        hardCodedPool1.poolWalletAddress,
+        hardCodedPool1.baseAssetId
+      );
+      const amount = baseAssetFundsCurrentlyInPoolsBeforeTx.free.sub(new BN(Pica(10_000).toString()));
+      const {
+        data: [
+          resultPoolId,
+          resultWho,
+          resultBaseAsset,
+          resultQuoteAsset,
+          resultBaseAmount,
+          resultQuoteAmount,
+          resultFee
+        ]
+      } = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.buy(hardCodedPool1.poolId, 130, { assetId: 4, amount: amount }, false)
+      );
+    });
+
+    it("#4.14 I can swap a huge amount with very high slippage.", async function () {
+      const quoteAssetFundsCurrentlyInPoolsBeforeTx = await api.query.tokens.accounts(
+        hardCodedPool1.poolWalletAddress,
+        hardCodedPool1.quoteAssetId
+      );
+      const amount = quoteAssetFundsCurrentlyInPoolsBeforeTx.free.sub(new BN(USDT(1).toString()));
+      const {
+        data: [
+          resultPoolId,
+          resultWho,
+          resultBaseAsset,
+          resultQuoteAsset,
+          resultBaseAmount,
+          resultQuoteAmount,
+          resultFee
+        ]
+      } = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(
+          hardCodedPool1.poolId,
+          { assetId: hardCodedPool1.quoteAssetId, amount: amount },
+          {
+            assetId: hardCodedPool1.baseAssetId,
+            amount: 0
+          },
+          false
+        )
+      );
+    });
+
+    it("#4.17 I can buy in the pool with 0 fees & pay 0 fees.");
+
+    it("#4.18 I can swap in the pool with 0 fees & pay 0 fees.");
+  });
+
+  describe("3. Removing liquidity", function () {
+    it("#3.1  I can not remove more liquidity than the amount equivalent to my LP token amount.", async function () {
+      const poolQuery = await api.query.pablo.pools(0);
+      const lpTokenId = poolQuery.unwrap().asDualAssetConstantProduct.lpToken;
+      const lpAmount = <OrmlTokensAccountData>await api.query.tokens.accounts(walletLpProvider1.publicKey, lpTokenId);
+      const amountToRemove = lpAmount.free.muln(1.5);
+      const assets = api.createType("BTreeMap<u128, u128>", {
+        "4": Pica(BigInt(amountToRemove.toString())),
+        "130": USDT(BigInt(amountToRemove.toString()))
+      });
+
+      const exc = await sendAndWaitForSuccess(
+        api,
+        walletLpProvider1,
+        api.events.pablo.LiquidityRemoved.is,
+        api.tx.pablo.removeLiquidity(hardCodedPool1.poolId, amountToRemove, assets)
+      ).catch(exc => exc);
+      expect(exc.toString()).to.contain("pablo.CannotRespectMinimumRequested");
+    });
+
+    it("#3.2  I can not remove liquidity amounts of 0.", async function () {
```

Comment:
So shall we raise this as a bug as it allows people to remove liq with amounts of 0. Seems not of a big deal but still something to consider I would say.

Reply:
Don't think so. It will remove asset amounts of `0` anyway. And a user would loose the tx fee.